### PR TITLE
Make buffers have a length

### PR DESCRIPTION
**Pull request by martindurant**

### CODE REVIEW

**Comment by martindurant** [comment]
(this is needed by the fsspec integration, although I could work around it in obstore instead; this ought to be done anyway, though)

File: pyo3-arrow/src/buffer.rs [hunk 1]
```diff
@@ -80,6 +80,10 @@ impl PyArrowBuffer {
         PyBytes::new_bound(py, &self.0)
     }
 
+    fn __len__(&self) -> PyResult<usize> {
+        Ok(self.0.len())
+    }
```
Comment (kylebarron) [inline pyo3-arrow/src/buffer.rs:85]: This doesn't need to be fallible.
```suggestion
    fn __len__(&self) -> usize {
        self.0.len()
    }
```

**Review by kylebarron — APPROVED** [review]
This lgtm. Any reason why it's still in draft?

**Comment by martindurant** [comment]
I had trouble building it locally (need to update a bunch of stuff - I don't normally use poetry), so this was test-by-ci. I thought it was probably simple enough to work.

**Comment by kylebarron** [comment]
> I don't normally use poetry

This repo uses `uv` now!

**Comment by kylebarron** [comment]
It just needed the python typing to be updated as well, but now it's good

**Pull request merged**